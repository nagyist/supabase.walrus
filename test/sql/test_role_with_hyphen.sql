select 1 from pg_create_logical_replication_slot('realtime', 'wal2json', false);

create role "has-hyphen" nologin noinherit;
grant usage on schema public to "has-hyphen";
alter default privileges in schema public grant all on tables to "has-hyphen";
alter default privileges in schema public grant all on functions to "has-hyphen";
alter default privileges in schema public grant all on sequences to "has-hyphen";



create table public.notes(
    id int primary key
);

insert into realtime.subscription(subscription_id, entity, claims)
select
    seed_uuid(1),
    'public.notes',
    jsonb_build_object(
        'role', 'has-hyphen',
        'email', 'example@example.com',
        'sub', seed_uuid(1)::text
    );

select clear_wal();
insert into public.notes(id) values (1);

select
    rec,
    is_rls_enabled,
    subscription_ids,
    errors
from
   walrus;


drop table public.notes;
select pg_drop_replication_slot('realtime');
truncate table realtime.subscription;
