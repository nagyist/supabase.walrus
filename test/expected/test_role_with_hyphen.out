select 1 from pg_create_logical_replication_slot('realtime', 'wal2json', false);
 ?column? 
----------
        1
(1 row)

create role "has-hyphen" nologin noinherit;
create schema private;
grant usage on schema private to "has-hyphen";
alter default privileges in schema private grant all on tables to "has-hyphen";
alter default privileges in schema private grant all on functions to "has-hyphen";
alter default privileges in schema private grant all on sequences to "has-hyphen";
create table private.notes(
    id int primary key
);
create policy rls_note_select
on private.notes
to "has-hyphen"
using (true);
alter table private.notes enable row level security;
insert into realtime.subscription(subscription_id, entity, claims)
select
    seed_uuid(1),
    'private.notes',
    jsonb_build_object(
        'role', 'has-hyphen',
        'email', 'example@example.com',
        'sub', seed_uuid(1)::text
    );
select clear_wal();
 clear_wal 
-----------
 
(1 row)

insert into private.notes(id) values (1);
select
    rec,
    is_rls_enabled,
    subscription_ids,
    errors
from
   walrus;
                        rec                         | is_rls_enabled |            subscription_ids            | errors 
----------------------------------------------------+----------------+----------------------------------------+--------
 {                                                 +| t              | {f4539ebe-c779-5788-bbc1-2421ffaa8954} | {}
     "type": "INSERT",                             +|                |                                        | 
     "table": "notes",                             +|                |                                        | 
     "record": {                                   +|                |                                        | 
         "id": 1                                   +|                |                                        | 
     },                                            +|                |                                        | 
     "schema": "private",                          +|                |                                        | 
     "columns": [                                  +|                |                                        | 
         {                                         +|                |                                        | 
             "name": "id",                         +|                |                                        | 
             "type": "int4"                        +|                |                                        | 
         }                                         +|                |                                        | 
     ],                                            +|                |                                        | 
     "commit_timestamp": "2000-01-01T08:01:01.000Z"+|                |                                        | 
 }                                                  |                |                                        | 
(1 row)

drop table private.notes;
drop schema private;
select pg_drop_replication_slot('realtime');
 pg_drop_replication_slot 
--------------------------
 
(1 row)

truncate table realtime.subscription;
